applications/nfc/nfc.c

```diff
@@ -224,6 +224,21 @@ void nfc_show_loading_popup(void* context, bool show) {
 }
 
 int32_t nfc_app(void* p) {
+    if(!furi_hal_nfc_is_init()) {
+        // No connection to the chip, show an error screen
+        DialogsApp* dialogs = furi_record_open(RECORD_DIALOGS);
+        DialogMessage* message = dialog_message_alloc();
+        dialog_message_set_text(
+            message,
+            "Error!\nNFC chip failed to start\n\n\nSend a photo of this to:\nsupport@flipperzero.one",
+            0,
+            0,
+            AlignLeft,
+            AlignTop);
+        dialog_message_show(dialogs, message);
+        return 0;
```

Comment:
Memory leak!
After dialog_message_show() you need
1) Free message: dialog_message_free(message);
2) Close record: furi_record_close(RECORD_DIALOGS);

Reply:
Also, I would prefer to take this to separate function
static bool nfc_is_hal_ready() {...}

and in nfc_app(){
if(!nfc_is_hal_ready()) return 0

Reply:
Fixed!